module/platform/Webots/src/Webots.cpp

```diff
@@ -0,0 +1,257 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2021 NUbots <nubots@nubots.net>
+ */
+
+#include "Webots.hpp"
+
+#include <chrono>
+#include <fmt/format.h>
+
+#include "extension/Configuration.hpp"
+
+#include "message/motion/ServoTarget.hpp"
+#include "message/output/CompressedImage.hpp"
+#include "message/platform/darwin/DarwinSensors.hpp"
+#include "message/platform/webots/ConnectRequest.hpp"
+#include "message/platform/webots/messages.hpp"
+#include "message/support/GlobalConfig.hpp"
+
+// Include headers needed for TCP connection
+extern "C" {
+#include <netdb.h>      /* definition of gethostbyname */
+#include <netinet/in.h> /* definition of struct sockaddr_in */
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <unistd.h> /* definition of close */
+}
+
+namespace module::platform {
+
+using extension::Configuration;
+using message::motion::ServoTargets;
+using message::output::CompressedImage;
+using message::platform::darwin::DarwinSensors;
+
+using message::platform::webots::ActuatorRequests;
+using message::platform::webots::ConnectRequest;
+using message::platform::webots::Message;
+using message::platform::webots::MotorPosition;
+using message::platform::webots::MotorTorque;
+using message::platform::webots::MotorVelocity;
+using message::platform::webots::SensorMeasurements;
+
+using message::support::GlobalConfig;
+
+
+int Webots::tcpip_connect(const std::string& server_name, const std::string& port) {
+    // Hints for the connection type
+    addrinfo hints;
+    memset(&hints, 0, sizeof(addrinfo));  // Defaults on what we do not explicitly set
+    hints.ai_family   = AF_UNSPEC;        // IPv4 or IPv6
+    hints.ai_socktype = SOCK_STREAM;      // TCP
+
+    // Store the ip address information that we will connect to
+    addrinfo* address;
+
+    int error;
+    if ((error = getaddrinfo(server_name.c_str(), port.c_str(), &hints, &address)) != 0) {
+        log<NUClear::ERROR>(fmt::format("Cannot resolve server name: {}. Error {}. Error code {}",
+                                        server_name,
+                                        gai_strerror(error),
+                                        error));
+        return -1;
+    }
+
+    // Loop through the linked list of potential options for connecting. In order of best to worst.
+    for (addrinfo* addr_ptr; addr_ptr != NULL; addr_ptr = addr_ptr->ai_next) {
```

Comment:
```suggestion
    for (addrinfo* addr_ptr = address; addr_ptr != NULL; addr_ptr = addr_ptr->ai_next) {
```
addr_ptr was instantiated, but not assigned, need to assign it to the beginning of the list.